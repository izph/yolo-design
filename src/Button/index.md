---
title: Button 按钮
nav:
  title: 组件
  path: /components
group:
  title: 通用
  order: 1
---

# Button 按钮

## Demo

```tsx
import React from 'react';
import { Button } from 'yolo-ui';

export default () => {
  return (
    <>
      <Button>按钮</Button>
      <Button type="primary">按钮</Button>
      <Button type="link">按钮</Button>
    </>
  );
};
```

## API

| 属性 |   说明   |                     类型                      | 默认值 |
| :--: | :------: | :-------------------------------------------: | :----: |
| type | 警告类型 | `info`、`success`、`error`、`warning`，非必填 | `info` |
