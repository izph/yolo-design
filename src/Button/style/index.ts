import '../../global.less';
import './index.less';
