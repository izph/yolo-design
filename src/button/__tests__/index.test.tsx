import React from 'react';
import { render } from '@testing-library/react';
import Button from '../index';

// test("button test case", function () {
//     const button = render(<Button>Test</Button>);
//     const element = button.queryByText('Test');
//     expect(element).toBeTruthy();
//     // 组件是否在文档中
//     expect(element).toBeInTheDocument();
// })

// describe将测试用例分类   it和test是一样的
describe('test Button Component', function () {
  // 测试不同类型的button
  it('should render Buttons of different type correctly', () => {
    const wrapper = render(
      <div>
        <Button>default</Button>
        <Button type="primary">primary</Button>
        <Button type="info">info</Button>
        <Button type="warning">warning</Button>
        <Button type="danger">danger</Button>
        <Button type="dashed">dashed</Button>
        <Button type="link">text</Button>
      </div>,
    );
    //生成 __snapshots__
    expect(wrapper).toMatchSnapshot();
  });

  // 单独测试link
  it('should render a link when btnType equals link and href is provided', () => {
    // const wrapper = shallow(
    //     <Button type="link" href="https://github.com/Jacky-Summer">
    //         Link
    //     </Button>,
    // )
    // const element = wrapper.first()
    // expect(element.name()).toEqual('a')
    // expect(element.hasClass('mk-btn mk-btn-link'))
  });

  // 测试不同大小的button
  it('should render Buttons of different size correctly', () => {
    const wrapper = render(
      <div>
        <Button type="primary" size="lg">
          lg尺寸
        </Button>
        <Button type="primary" size="md">
          md尺寸
        </Button>
        <Button type="primary" size="sm">
          sm尺寸
        </Button>
      </div>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render disabled button when disabled set to true', () => {
    // const wrapper = shallow(<Button disabled>Disabled Button</Button>)
    // const element = wrapper.find('button')
    // expect(element.prop('disabled')).toBeTruthy()
    // element.simulate('click')
    // expect(disabledProps.onClick).not.toHaveBeenCalled()
  });

  it('should not render as link button when href is undefined', async () => {
    // const wrapper = shallow(
    //     <Button type="link" href={undefined}>
    //         button
    //     </Button>,
    // )
    // expect(wrapper.find('a').exists()).toBeFalsy()
    // expect(wrapper.find('button').exists()).toBeTruthy()
  });
});
