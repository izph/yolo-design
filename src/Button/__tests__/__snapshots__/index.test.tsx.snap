// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button Component should render Buttons of different size correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="yolo-btn yolo-btn-primary yolo-btn-lg"
        >
          lg尺寸
        </button>
        <button
          class="yolo-btn yolo-btn-primary yolo-btn-md"
        >
          md尺寸
        </button>
        <button
          class="yolo-btn yolo-btn-primary yolo-btn-sm"
        >
          sm尺寸
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="yolo-btn yolo-btn-primary yolo-btn-lg"
      >
        lg尺寸
      </button>
      <button
        class="yolo-btn yolo-btn-primary yolo-btn-md"
      >
        md尺寸
      </button>
      <button
        class="yolo-btn yolo-btn-primary yolo-btn-sm"
      >
        sm尺寸
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button Component should render Buttons of different type correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="yolo-btn yolo-btn-default"
        >
          default
        </button>
        <button
          class="yolo-btn yolo-btn-primary"
        >
          primary
        </button>
        <button
          class="yolo-btn yolo-btn-info"
        >
          info
        </button>
        <button
          class="yolo-btn yolo-btn-warning"
        >
          warning
        </button>
        <button
          class="yolo-btn yolo-btn-danger"
        >
          danger
        </button>
        <button
          class="yolo-btn yolo-btn-dashed"
        >
          dashed
        </button>
        <button
          class="yolo-btn yolo-btn-link"
        >
          text
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="yolo-btn yolo-btn-default"
      >
        default
      </button>
      <button
        class="yolo-btn yolo-btn-primary"
      >
        primary
      </button>
      <button
        class="yolo-btn yolo-btn-info"
      >
        info
      </button>
      <button
        class="yolo-btn yolo-btn-warning"
      >
        warning
      </button>
      <button
        class="yolo-btn yolo-btn-danger"
      >
        danger
      </button>
      <button
        class="yolo-btn yolo-btn-dashed"
      >
        dashed
      </button>
      <button
        class="yolo-btn yolo-btn-link"
      >
        text
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
